geode-wan/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/AlterRegionCommandDUnitTest.java

```diff
@@ -343,4 +346,76 @@ public void alterPartitionPersistentRegionWithParallelNonPersistentGatewaySender
       assertThat(regionConfig.getRegionAttributes().getGatewaySenderIds()).isBlank();
     });
   }
+
+  @Test
+  public void alterReplicateRegionWithParallelGatewaySenderShouldFailAndDoNotPersistChangesIntoTheClusterConfigurationService() {
+    addIgnoredException(GatewaySenderConfigurationException.class);
+    addIgnoredException("could not get remote locator information");
+    String regionName = testName.getMethodName();
+    String gatewaySenderId = testName.getMethodName() + "_parallelGatewaySender";
+
+    gfsh.executeAndAssertThat(
+        "create gateway-sender --parallel=true --enable-persistence=false --remote-distributed-system-id=2 --id="
+            + gatewaySenderId)
+        .statusIsSuccess().doesNotContainOutput("Did not complete waiting");
+    gfsh.executeAndAssertThat("create region --type=REPLICATE --name=" + regionName)
+        .statusIsSuccess();
+    locator.waitUntilRegionIsReadyOnExactlyThisManyServers(SEPARATOR + regionName, 1);
+
+    // Associate the gateway-sender
+    gfsh.executeAndAssertThat(
+        "alter region --name=" + regionName + " --gateway-sender-id=" + gatewaySenderId)
```

Comment:
Another place to possibly use `CommandStringBuilder`.

Reply:
Done!.